### PR TITLE
Create temp table with same batching policy as destination table

### PR DESCRIPTION
**Pull request by yihezkel**
#### Pull Request Description

Create temp table with same batching policy as destination table

---

#### Future Release Comment

**Fixes:**
- Create temp table with same batching policy as destination table

### CODE REVIEW

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoBlobStorageUtils.scala [hunk 1]
```diff
@@ -37,7 +39,8 @@ object KustoBlobStorageUtils {
     val blobContainer = blobClient.getContainerReference(container)
     val blobsWithPrefix = blobContainer.listBlobs(directory)
 
-    blobsWithPrefix.forEach(blob => {
+    // foreach and not forEach as in scala 12
+    blobsWithPrefix.foreach(blob => {
```
Comment (ohadbitt) [inline connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoBlobStorageUtils.scala:43]: ? 
revert

File: docs/KustoSink.md [hunk 2]
```diff
@@ -116,7 +116,7 @@ that is using it. Please verify the following first:
 
 Write performance depends on multiple factors, such as scale of both Spark and Kusto clusters.
 Regarding Kusto target cluster configuration, one of the factors that impacts performance and latency 
-is [Ingestion Batching Policy](https://docs.microsoft.com/azure/kusto/concepts/batchingpolicy). The default policy 
+is [Ingestion Batching Policy](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/batchingpolicy). The default policy
```
Comment (ohadbitt) [inline docs/KustoSink.md:119]:
```suggestion
is [Ingestion Batching Policy](https://docs.microsoft.com/azure/data-explorer/kusto/management/batchingpolicy). The default policy 
```

File: pom.xml [hunk 3]
```diff
@@ -11,10 +11,10 @@
         <revision>3.0.0</revision>
 
         <!-- Spark dependencies -->
-        <scala.version.major>2.12</scala.version.major>
-        <scala.version.minor>11</scala.version.minor>
-        <spark.version.major>3.0</spark.version.major>
-        <spark.version.minor>1</spark.version.minor>
+        <scala.version.major>2.11</scala.version.major>
```
Comment (ohadbitt) [inline pom.xml:14]: revert

**Review by ohadbitt — COMMENTED** [review]
i left comments - all i think regarding you using scala 11 and not 12 - this project main branch is 12 and branch 2.4 is 11 
anyway theres no real point to this PR - as i said it doesnt work

File: connector/src/main/scala/com/microsoft/kusto/spark/authentication/DeviceAuthentication.scala [hunk 4]
```diff
@@ -21,10 +21,12 @@ class DeviceAuthentication (val cluster: String, val authority:String) extends a
   }
 
   def acquireNewAccessTokenAsync(): CompletableFuture[IAuthenticationResult] = {
-    val deviceCodeConsumer: Consumer[DeviceCode] = (deviceCode: DeviceCode) => {
-      this.deviceCode = Some(deviceCode)
-      expiresAt = Some(System.currentTimeMillis + (deviceCode.expiresIn() * 1000))
-      println(deviceCode.message())
+    val deviceCodeConsumer: Consumer[DeviceCode] = new Consumer[DeviceCode] {
```
[the remark this answers is not in the document]
Comment (yihezkel) [reply connector/src/main/scala/com/microsoft/kusto/spark/authentication/DeviceAuthentication.scala:24]: I apparently built this branch atop 2.4, so most of these weird changes were unintended. I'll abandon this and recreate just my changes atop master.

**Pull request closed without merging**